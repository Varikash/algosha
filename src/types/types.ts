import { ElementStates } from "./element-states";

export type StringElement = {
  value: string;
  state: ElementStates;
}