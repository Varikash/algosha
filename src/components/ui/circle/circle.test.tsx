import React from 'react';
import renderer from 'react-test-renderer';
import { Circle } from './circle';
import { ElementStates } from "../../../types/element-states";

describe('Circle component', () => {
  it("Рендерит circle с буквой", () => {
    const tree = renderer.create(<Circle letter="J" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Рендерит circle без буквы", () => {
    const tree = renderer.create(<Circle letter="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("Рендерит circle с head", () => {
    const tree = renderer.create(<Circle head="head" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("Рендерит circle с react-элементом в head", () => {
    const tree = renderer
      .create(<Circle head={<Circle isSmall={true} />} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("Рендерит circle с tail", () => {
    const tree = renderer.create(<Circle tail="tail" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Рендерит circle с react-элементом в tail', () => {
    const tree = renderer.create(<Circle tail={<Circle isSmall={true} />} />).toJSON();
    expect(tree).toMatchSnapshot();
  })
  
  it("Рендерит circle с index", () => {
    const tree = renderer.create(<Circle index={7} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("Рендерит circle с пропом isSmall ===  true", () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("Рендерит circle в состоянии default", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("Рендерит circle в состоянии changing", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("Рендерит circle в состоянии modified", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });





})