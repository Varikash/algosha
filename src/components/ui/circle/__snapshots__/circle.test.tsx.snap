// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component Рендерит circle без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component Рендерит circle в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   changing"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component Рендерит circle в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component Рендерит circle в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   modified"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component Рендерит circle с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-cy="head"
  >
    head
  </div>
  <div
    className="circle   default"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component Рендерит circle с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    7
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component Рендерит circle с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
      />
      <div
        className="circle  small default"
        data-cy="circleStyle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component Рендерит circle с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
      />
      <div
        className="circle  small default"
        data-cy="circleStyle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-cy="letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component Рендерит circle с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-cy="tail"
  >
    tail
  </div>
</div>
`;

exports[`Circle component Рендерит circle с буквой 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle   default"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    >
      J
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component Рендерит circle с пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  small default"
    data-cy="circleStyle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-cy="letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;
